### PR TITLE
Introduce reranker-cohere module

### PR DESCRIPTION
**Pull request by CShorten**
### What's being changed:
What's being changed:
Cross Encoder Re-Rankers in Weaviate! (From Cohere API)

To configure:
COHERE_APIKEY: 'cohere api key'
ENABLE_MODULES: 'reranker-cohere'

To query:
```graphql
{
  Get {
    Product(limit:50){
      content
      _additional {
        rerank(
          property: "content"
          query: "what is ref2vec?"
        ) {
        score
        }
      }
  }
}
```

Review checklist
 Documentation has been updated, if necessary. Link to changed documentation:
 Chaos pipeline run or not necessary. Link to pipeline:
 All new code is covered by tests where it is reasonable.
 Performance tests have been run or not necessary.

### Review checklist

- [ ] Documentation has been updated, if necessary. Link to changed documentation:
- [X] Chaos pipeline run or not necessary. Link to pipeline:
- [X] All new code is covered by tests where it is reasonable.
- [X] Performance tests have been run or not necessary.

### CODE REVIEW

File: adapters/handlers/rest/configure_api.go [hunk 1]
```diff
@@ -512,14 +514,22 @@ func registerModules(appState *state.State) error {
 			Debug("enabled module")
 	}
 
-	if _, ok := enabledModules[modrerankertransformers.Name]; ok {
+	if _, ok := enabledModules["reranker-transformers"]; ok {
```
Comment (antas-marcin) [inline adapters/handlers/rest/configure_api.go:517]: you can change "reranker-transformers" to `modrerankercohere.Name`

Comment (antas-marcin) [reply adapters/handlers/rest/configure_api.go:517]: as it was before

File: modules/reranker-cohere/clients/ranker.go [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+	}
+
+	if res.StatusCode > 399 {
+		return nil, errors.Errorf("fail with status %d: %s", res.StatusCode,
+			"An error occurred") // Modify as per your error handling
+	}
+	/*
+		Example response
+		{"id":"862118d6-516e-4d5e-b6d1-c67b8643e58e","results":[{"index":0,"relevance_score":0.98415464}],"meta":{"api_version":{"version":"1"}}}
+	*/
```
Comment (antas-marcin) [inline modules/reranker-cohere/clients/ranker.go:106]: you can remove this commented part

File: modules/reranker-cohere/clients/ranker_meta.go [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+import (
+	"context"
+	"encoding/json"
+	"io"
+	"net/http"
+
+	"github.com/pkg/errors"
+)
+
+func (s *client) MetaInfo() (map[string]interface{}, error) {
```
Comment (antas-marcin) [inline modules/reranker-cohere/clients/ranker_meta.go:23]: there is no `meta` endpoint, this part should be removed. Look [here](https://github.com/weaviate/weaviate/blob/master/modules/generative-cohere/clients/cohere_meta.go) how it should be implemented

File: modules/reranker-cohere/config/class_settings_test.go [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+// \ \ /\ / / _ \/ _` \ \ / / |/ _` | __/ _ \
+//  \ V  V /  __/ (_| |\ V /| | (_| | ||  __/
+//   \_/\_/ \___|\__,_| \_/ |_|\__,_|\__\___|
+//
+//  Copyright © 2016 - 2023 Weaviate B.V. All rights reserved.
+//
+//  CONTACT: hello@weaviate.io
+//
+
+package config
```
Comment (antas-marcin) [inline modules/reranker-cohere/config/class_settings_test.go:12]: if there are no tests for `class_settings` then I would advise to remove this file or add the tests :)

File: modules/reranker-cohere/additional/rank/rank_graphql_field.go [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+				DefaultValue: nil,
+			},
+			"property": &graphql.ArgumentConfig{
+				Description:  "Property to rank from",
+				Type:         graphql.String,
+				DefaultValue: nil,
+			},
+		},
+		Type: graphql.NewList(graphql.NewObject(graphql.ObjectConfig{
+			Name: fmt.Sprintf("%sAdditionalSummary", classname),
```
[the remark this answers is not in the document]
Comment (CShorten) [reply modules/reranker-cohere/additional/rank/rank_graphql_field.go:35]: Oops! Thanks for the catch!

File: modules/reranker-cohere/clients/ranker_meta.go (hunk 3 above)
Comment (CShorten) [reply modules/reranker-cohere/clients/ranker_meta.go:23]: Ah fantastic, thanks for the clear example!

File: modules/reranker-cohere/clients/ranker_test.go [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+		/*
+			res, err := c.Rank(context.Background(), "prop", "I work at Apple")
+
+			assert.Nil(t, err)
+			assert.Equal(t, ent.RankResult{
+				Query:             "Where do I work?",
+				RankPropertyValue: "I work at Apple",
+				Score:             0.15,
+			}, *res)
+		*/
```
Comment (antas-marcin) [inline modules/reranker-cohere/clients/ranker_test.go:47]: please remove commented code

File: modules/reranker-cohere/clients/ranker_test.go [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+			},
+		})
+		defer server.Close()
+		//c := New(server.URL, nullLogger())
+		/*
+			_, err := c.Rank(context.Background(), "I work at Apple")
+
+			require.NotNil(t, err)
+			assert.Contains(t, err.Error(), "some error from the server")
+		*/
```
Comment (antas-marcin) [inline modules/reranker-cohere/clients/ranker_test.go:64]: please remove commented code

File: modules/reranker-cohere/clients/ranker_test.go [hunk 8]
```diff
@@ -0,0 +1,85 @@
...
+}
+
+func (f *testCrossRankerHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	assert.Equal(f.t, "/rerank", r.URL.String())
+	assert.Equal(f.t, http.MethodPost, r.Method)
+
+	/*if f.res.Meta != "" {
+		w.WriteHeader(500)
+	}
+	*/
```
Comment (antas-marcin) [inline modules/reranker-cohere/clients/ranker_test.go:81]: please remove commented code

File: adapters/handlers/rest/configure_api.go [hunk 9]
```diff
@@ -512,14 +514,22 @@ func registerModules(appState *state.State) error {
...
 		appState.Modules.Register(modrerankertransformers.New())
 		appState.Logger.
 			WithField("action", "startup").
 			WithField("module", modrerankertransformers.Name).
 			Debug("enabled module")
 	}
 
+	if _, ok := enabledModules["reranker-cohere"]; ok {
```
[the remark this answers is not in the document]
Comment (antas-marcin) [reply adapters/handlers/rest/configure_api.go:525]: did you push this change? I don't see that this part changed

[the remark this answers is not in the document]
Comment (CShorten) [reply adapters/handlers/rest/configure_api.go:525]: Yes I believe so -- it is changed on line 529

**Pull request merged**